From 60f872f8174d5267cf116ea738e1e37b3eeb0fd8 Mon Sep 17 00:00:00 2001
From: Pavel Dubrova <pashadubrova@gmail.com>
Date: Sat, 6 Nov 2021 21:26:22 +0200
Subject: [PATCH 063/175] soc: qcom: Add soc information for SDM845

Add SOC information like Chip ID, macros and dummy socinfo
definitions for SDM845 SoC.

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>
---
 drivers/soc/qcom/socinfo.c | 7 +++++++
 include/soc/qcom/socinfo.h | 4 ++++
 2 files changed, 11 insertions(+)

diff --git a/drivers/soc/qcom/socinfo.c b/drivers/soc/qcom/socinfo.c
index d1b52f7e548a..df794df11d29 100644
--- a/drivers/soc/qcom/socinfo.c
+++ b/drivers/soc/qcom/socinfo.c
@@ -341,6 +341,9 @@ static struct msm_soc_info cpu_of_id[] = {
 	/* SDM660 ID */
 	[317] = {MSM_CPU_SDM660, "SDM660"},
 
+	/* SDM845 ID */
+	[321] = {MSM_CPU_SDM845, "SDM845"},
+
 	/* sm8150 ID */
 	[339] = {MSM_CPU_SM8150, "SM8150"},
 
@@ -1546,6 +1549,10 @@ static void * __init setup_dummy_socinfo(void)
 		dummy_socinfo.id = 317;
 		strlcpy(dummy_socinfo.build_id, "sdm660 - ",
 		sizeof(dummy_socinfo.build_id));
+	} else if (early_machine_is_sdm845()) {
+		dummy_socinfo.id = 321;
+		strlcpy(dummy_socinfo.build_id, "sdm845 - ",
+		sizeof(dummy_socinfo.build_id));
 	} else if (early_machine_is_sm8150()) {
 		dummy_socinfo.id = 339;
 		strlcpy(dummy_socinfo.build_id, "sm8150 - ",
diff --git a/include/soc/qcom/socinfo.h b/include/soc/qcom/socinfo.h
index 70aa770f6da7..b858ec606b78 100644
--- a/include/soc/qcom/socinfo.h
+++ b/include/soc/qcom/socinfo.h
@@ -151,6 +151,8 @@ enum socinfo_parttype {
 	of_flat_dt_is_compatible(of_get_flat_dt_root(), "qcom,sdmmagpie")
 #define early_machine_is_sdm660()	\
 	of_flat_dt_is_compatible(of_get_flat_dt_root(), "qcom,sdm660")
+#define early_machine_is_sdm845()	\
+	of_flat_dt_is_compatible(of_get_flat_dt_root(), "qcom,sdm845")
 #define early_machine_is_bengal_iot()	\
 	of_flat_dt_is_compatible(of_get_flat_dt_root(), "qcom,bengal-iot")
 #define early_machine_is_bengalp_iot()	\
@@ -210,6 +212,7 @@ enum socinfo_parttype {
 #define early_machine_is_sdxprairie()	0
 #define early_machine_is_sdmmagpie()	0
 #define early_machine_is_sdm660()	0
+#define early_machine_is_sdm845()	0
 #define early_machine_is_bengal_iot()	0
 #define early_machine_is_bengalp_iot()	0
 #define early_machine_is_msm8937()	0
@@ -241,6 +244,7 @@ enum msm_cpu {
 	MSM_CPU_8084,
 	MSM_CPU_8996,
 	MSM_CPU_SDM660,
+	MSM_CPU_SDM845,
 	MSM_CPU_SM8150,
 	MSM_CPU_SA8150,
 	MSM_CPU_KONA,
-- 
2.48.1

